Cabal/src/Distribution/Simple/Test/ExeV10.hs

```diff
@@ -171,11 +170,9 @@ runTest pkg_descr lbi clbi flags suite = do
   notice verbosity $ summarizeSuiteFinish suiteLog
 
   when isCoverageEnabled $
-    case PD.library pkg_descr of
-      Nothing ->
-        dieWithException verbosity TestCoverageSupport
-      Just library ->
-        markupTest verbosity lbi distPref (prettyShow $ PD.package pkg_descr) suite library
+    if null $ testCoverageDistPrefs flags
+      then dieWithException verbosity TestCoverageSupport
```

Comment:
I suppose I'm conflating again `cabal-install` and `Cabal`.

The idea in using the same error is that we want to report the same information to the user:
if there are no library components in the package, cabal-install will pass empty `testCoverageDistPrefs`, and then `Cabal` is reporting that there are no library components because it received empty flags.

Perhaps the right way would be
* `cabal-install` checks for existing components 
* reports errors if empty
* passes flags to cabal
* cabal reports "no coverage dist prefs when coverage enabled"
?